# Bioinformatics resources

For bioinformatics and data analysis in genomics, there are some useful foundational skills and tools, as well as some specific to particular analyses. This is meant to be a resource of links to tutorials on many of these topics. A few courses also run that focus on these skills in particular, and have sets of resources.



## Courses

These courses focus on bioinformatics analyses. There are links for all the materials and they're all open and freely available.

- [NGS course on Next-Generation Sequence analysis](http://angus.readthedocs.io/en/2016/)
- [EDAMAME course on microbial metagenomics](https://github.com/edamame-course/2016-tutorials/wiki/Learning-Goals)

## Foundational skills

### Data organization

The foundational element of any data analysis project is organizing your data so it can be used in any of these programs

- [Data Carpentry data organization in spreadsheets](http://www.datacarpentry.org/spreadsheet-ecology-lesson/)

### Introduction to R

Some resources for learning R

- [Data Carpentry R for data analysis](http://www.datacarpentry.org/R-ecology-lesson/)
- [Software Carpentry Programming with R](http://swcarpentry.github.io/r-novice-inflammation/)
- [Data Camp Introduction to R](https://www.datacamp.com/courses/free-introduction-to-r)
- [Coursera R programming course](https://www.coursera.org/learn/r-programming)

### Introduction to Python

Some resources for learning Python

- [Software Carpentry Programming with Python](http://swcarpentry.github.io/python-novice-inflammation/)
- [Data Carpentry Python for data analysis](http://www.datacarpentry.org/python-ecology-lesson/)
-

### Introduction to the command line

The command line is important for running most bioinformatics programs

- [Software Carpentry Introduction to the command line](http://swcarpentry.github.io/shell-novice/)
- [Data Carpentry command line for genomics, including automation](https://github.com/datacarpentry/shell-genomics/blob/gh-pages/index.md)

### Using Amazon EC2 instances

Many analyses are too computationally intensive to run on your own computer. Another option is to use cloud resources. There are many types, including the HPC at your own university. Information for Amazon EC2 cloud computing is just hte one outlined here. It's currently the most frequently used and there are things set up here for bioinformatics.

- [NGS Getting started with EC2](http://angus.readthedocs.io/en/2015/amazon/index.html)
- [EDAMAME Starting up EC2 instances](https://github.com/edamame-course/2015-tutorials/blob/master/final/2015-06-22-EC2_Startup.md)
- [EDAMAME Transfering data to EC2](https://github.com/edamame-course/2015-tutorials/blob/master/final/2015-06-22-EC2_Connection_FileTransfer.md)

### Introduction to github

Github is used for version control. It's a way to keep track of changes you make in your code and to share your code publicly if you wish. You can have private repositories as well, if you aren't yet ready to share your code.

- [Software Carpentry introduction to github](http://swcarpentry.github.io/git-novice/)

## Analysis specific tutorials

Building off the foundational skills these are tutorials on the more specific analyses. Some of the tutorials assume some knowledge of the command line.

### Sequence quality filtering

### BLAST databases

### mothur

- [Using mothur on an Amazon instance](http://mothur.org/wiki/Mothur_AMI)
- [mothur SOP](http://www.mothur.org/wiki/MiSeq_SOP)

### QIIME

- [Starting the QIIME AMI](https://github.com/edamame-course/Amplicon_Analysis/blob/master/final/QIIME0.md) from EDAMAME
- [QIIME intro tutorial](https://github.com/edamame-course/Amplicon_Analysis/blob/master/final/2016-07-13-QIIME1.md) from EDAMAME
- [QIIME working with OTU tables](https://github.com/edamame-course/Amplicon_Analysis/blob/master/final/2016-07-13-QIIME2.md) from EDAMAME
- [QIIME comparative diversity](https://github.com/edamame-course/Amplicon_Analysis/blob/master/final/2016-07-13-QIIME3.md) from EDAMAME

### RNASeq

### Variant calling

### Genome assembly

### Metagenome assembly

### Making phylogenetic trees with ARB

### Genome annotation

### Metagenome annotation
